compiler/codegen/src/symboltable.rs

```diff
@@ -158,7 +154,7 @@ impl Symbol {
     }
 
     pub fn is_bound(&self) -> bool {
-        self.is_assigned || self.is_parameter || self.is_imported || self.is_iter
+        !(self.flags & SymbolFlags::BOUND).is_empty()
```

Comment:
```suggestion
        self.flags.intersects(SymbolFlags::BOUND)
```

is easier to read than the negation on empty here.

Reply:
Code fix completed.